src/zephyrAdc/zephyrAdc.c

```diff
@@ -0,0 +1,127 @@
+/**
+ * @file      zephyrAdc.c
+ * @author    jbacon
+ * @date      2021-04-09
+ * @brief     ADC of ZephyrWrapper module
+ *
+ *            This file is the implementation of the ADC wrapper
+ * 
+ * @ingroup  zephyr-wrapper
+ * @{
+ */
+
+#include <zephyrAdc.h>
+
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+
+#include "zephyrCommon.h"
+
+LOG_MODULE_DECLARE(ZEPHYR_WRAP_MODULE_NAME);
+
+/**
+ * @brief The ADC reference voltage.
+ */
+#define ADC_REF_VOLT              3300
+
+/**
+ * @brief The ADC data structure.
+ */
+typedef struct adcCtrlData
+{
+  const struct device *adc;                                 /**< The zephyr ADC device. */
+  struct adc_channel_cfg chanCfgs[ADC_MAX_CHAN_CNT];        /**< The zephyr ADC channel configuration. */
```

Comment:
This should be allocated dynamically based on the used channel count. Which should come from the device tree.